User must log in with Google to view data (resolves #86)

mjkrause opened the pull request:
In order to contain accidental or malicious abuse of the export function in the `dev` instance of the Commons Boardwalk browser we now force the user to log in with Google in order to use the export function. If a user isn't logged in no data can be viewed in the Data Browser.

That way a user can be tracked in case of need.

That change is implemented in https://mkrause6.ucsc-cgp-dev.org.

Commit:
User must log in with Google to view data (resolves #86)

In order to contain accidental or malicious abuse of the export
function in a `dev` instance of the Commons Boardwalk Data Browser
we now force the user to log in with Google in order to view any
data, and to use the export function. If a user isn't logged in
no data can be viewed in the Data Browser.

That way a user can be tracked in case of need.

jessebrennan reviewed (changes requested):
Looks good! Now the docstring is out of date though. Can you update it? Specifically https://github.com/DataBiosphere/cgp-dashboard/pull/87/files#diff-3f41e546893dc64b71aaacad12cad815R434

mjkrause commented:
Hi Jesse,
I updated the docstring to reflect the changes. 

Also, I added my email to the whitelist of `commons/dev/whitelist`, and re-deployed `mkrause6.ucsc-cgp-dev.org`. I confirmed that a 401 is returned when I click tab _Data Browser_ if I'm not logged in. After I logged in, clicking the same tab returned a 204 (and I could see the data). 

Earlier I had tested only the condition that I was authorized, but my email wasn't on the whitelist.

I think all cases are covered now.
